go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -245,16 +264,73 @@ func newSharedGatewayOpenFlowManager(patchPort, macAddress, gwBridge, gwIntf str
 		dftFlows = append(dftFlows,
 			fmt.Sprintf("cookie=%s, priority=50, in_port=%s, ipv6, "+
 				"actions=ct(zone=%d, table=1)", defaultOpenFlowCookie, ofportPhys, config.Default.ConntrackZone))
+
+		physicalIP, err := util.MatchIPNetFamily(true, ips)
+		if err != nil {
+			return nil, fmt.Errorf("unable to determine IPv6 physical IP of host: %v", err)
+		}
+		// table 0, SVC Hairpin from OVN destined to local host, DNAT to host, send to table 4
+		dftFlows = append(dftFlows,
+			fmt.Sprintf("cookie=%s, priority=500, in_port=%s, ipv6, ipv6_dst=%s, ipv6_src=%s,"+
+				"actions=ct(commit,zone=%d,nat(dst=%s),table=3)",
```

Comment:
You are sending it to table 3 here, either that is correct for IPv6 (in which case the comment above is incorrect) or you overlooked that.

Reply:
yes this is wrong, it should be table 4. Nice catch